pkg/fqdn/service/service.go

```diff
@@ -135,52 +268,90 @@ func NewServer(endpointManager endpointmanager.EndpointManager, updateOnDNSMsg m
 
 // OnIPIdentityCacheChange is a method to receive the IP identity cache change events
 func (s *FQDNDataServer) OnIPIdentityCacheChange(modType ipcache.CacheModification, cidr types.PrefixCluster, oldHostIP, newHostIP net.IP, oldID *ipcache.Identity, newID ipcache.Identity, encryptKey uint8, k8sMeta *ipcache.K8sMetadata, endpointFlags uint8) {
-	s.identityToIPMutex.Lock()
-	defer s.identityToIPMutex.Unlock()
-
 	if cidr.ClusterID() != 0 {
 		return
 	}
+
+	txn := s.db.WriteTxn(s.identityToIPsTable)
```

Comment:
This is making me want to move the ipcache to an exported statedb table :-)

Reply:
This thought had crossed my mind a few times. I think that statedb in some ways was a formalization of some of the ideas we had when revamping the ipcache package.